<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>376</width>
    <height>407</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ljh2off Projectors GUI</string>
  </property>
  <property name="toolTip">
   <string>path of a single ljh file with pulses, other channel numbers will be found automatically</string>
  </property>
  <property name="statusTip">
   <string/>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QFrame" name="frame_2">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QPushButton" name="pulsePathButton">
        <property name="minimumSize">
         <size>
          <width>60</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>path a a single ljh file with pulses, other channel numbers will be found automatically</string>
        </property>
        <property name="statusTip">
         <string/>
        </property>
        <property name="whatsThis">
         <string/>
        </property>
        <property name="text">
         <string>Pulse Path</string>
        </property>
        <property name="flat">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="pulsePathLineEdit">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="frame_3">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QPushButton" name="noisePathButton">
        <property name="minimumSize">
         <size>
          <width>60</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>path of a single ljh file with noise records, other channel numbers will be found automatically</string>
        </property>
        <property name="statusTip">
         <string/>
        </property>
        <property name="whatsThis">
         <string/>
        </property>
        <property name="text">
         <string>Noise Path</string>
        </property>
        <property name="flat">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="noisePathLineEdit">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="frame">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Output File Name</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="outputFileNameBox">
        <property name="toolTip">
         <string>output filename, the default behavior will place it in the same directory as pulse_path</string>
        </property>
        <property name="placeholderText">
         <string>Default: [date]_run[X]_model.hdf5</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="replaceCheckbox">
        <property name="text">
         <string>Replace existing output files</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="frame_4">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="8" column="1">
       <widget class="QDoubleSpinBox" name="maxPulsesBox">
        <property name="toolTip">
         <string>maximum number of pulses to take a truncated SVD of, will affect speed</string>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="maximum">
         <double>10000000.000000000000000</double>
        </property>
        <property name="value">
         <double>4000.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QDoubleSpinBox" name="sigmaMaxDerivBox">
        <property name="toolTip">
         <string>passed to autocuts to determine pulses used to generate pulse model</string>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="maximum">
         <double>10000000.000000000000000</double>
        </property>
        <property name="value">
         <double>8.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Sigma Max Derivative</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QDoubleSpinBox" name="presamplesBox">
        <property name="toolTip">
         <string>ignore this many presample before the rising edge when calculating pretrigger_mean</string>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="maximum">
         <double>10000000.000000000000000</double>
        </property>
        <property name="value">
         <double>3.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Sigma Pretrigger RMS</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QDoubleSpinBox" name="cutoffATSBox">
        <property name="toolTip">
         <string>cutoff frequency for arrival time safe filter</string>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="decimals">
         <number>2</number>
        </property>
        <property name="maximum">
         <double>10000000.000000000000000</double>
        </property>
        <property name="value">
         <double>0.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QDoubleSpinBox" name="cutoff5lagBox">
        <property name="toolTip">
         <string>cutoff frequency for 5lag filter</string>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="decimals">
         <number>2</number>
        </property>
        <property name="maximum">
         <double>10000000.000000000000000</double>
        </property>
        <property name="value">
         <double>0.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QDoubleSpinBox" name="maxChansBox">
        <property name="toolTip">
         <string>stop after processing this many channels</string>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="maximum">
         <double>10000000.000000000000000</double>
        </property>
        <property name="value">
         <double>999999.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Number of Projectors (&gt;2)</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QDoubleSpinBox" name="nprojectorsBox">
        <property name="toolTip">
         <string>how many projectors to generate, must be &gt;=3</string>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="maximum">
         <double>10000000.000000000000000</double>
        </property>
        <property name="value">
         <double>5.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Cutoff Freq for 5lag (0=None)</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Presamples to Ignore</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QDoubleSpinBox" name="sigmaPtRmsBox">
        <property name="toolTip">
         <string>passed to autocuts to determine pulses used to generate pulse model, the default large value basically disables this cut, which causes the projectors to be able to model the non-flat pretrigger region better</string>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="maximum">
         <double>10000000.000000000000000</double>
        </property>
        <property name="value">
         <double>10000.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Cutoff Freq for ATS (0=None)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Max Channels</string>
        </property>
       </widget>
      </item>
      <item row="8" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Max Pulses</string>
        </property>
       </widget>
      </item>
      <item row="9" column="0">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Extra Basis Components for 5lag</string>
        </property>
       </widget>
      </item>
      <item row="9" column="1">
       <widget class="QDoubleSpinBox" name="extraBasis5lagBox">
        <property name="toolTip">
         <string>use this many basis components to improve 5 lag filter representation</string>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="maximum">
         <double>10000000.000000000000000</double>
        </property>
        <property name="value">
         <double>0.000000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="frame_7">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_6">
      <item>
       <widget class="QCheckBox" name="invertDataCheckbox">
        <property name="toolTip">
         <string>pass this flag for downward going pulses (eg RAVEN)</string>
        </property>
        <property name="text">
         <string>Invert data (eg. RAVEN)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="dontOptimiseCheckbox">
        <property name="toolTip">
         <string>simpler derivative like calculating, better for gamma data</string>
        </property>
        <property name="text">
         <string>Don't Optimize dp/dt (eg. gamma)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="noiseWeightingCheckbox">
        <property name="toolTip">
         <string>pass false to not apply noise weighting when making projectors</string>
        </property>
        <property name="text">
         <string>Apply Noise Weighting</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="startButton">
     <property name="text">
      <string>Make Projectors</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="plotProjectorsGroup">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>Channel</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="channelBox"/>
      </item>
      <item>
       <widget class="QPushButton" name="plotProjectorsButton">
        <property name="text">
         <string>Plot Projectors</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
